src/components/work/TripCard/TripCard.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import { Card, CardImg, CardBody, CardTitle, CardSubtitle, Row, Col, CardText, CardFooter } from 'reactstrap';
+import { Booking } from 'networking/bookings';
+import { getUserBookingDisplayStatus } from 'utils/bookingsDisplayStatus';
+import { formatAddress, formatGeolocationAddress } from 'utils/formatter';
+import { formatSingleDate } from 'utils/formatDate';
+
+enum ModalType {
+  CANCEL_BOOKING = 'CANCEL_BOOKING',
+  CONTACT_HOST = 'CONTACT_HOST',
+}
+
+interface Props {
+  booking: Booking;
+  handleModalAction: (modal: ModalType) => void;
+}
+
+const TripCard = ({ booking, handleModalAction }: Props) => {
+  const { checkInDate, checkOutDate, id, listing, status } = booking;
+  const { addressLine1, addressLine2, city, country, lat, lng, postalCode, state } = listing;
+  const displayStatus = getUserBookingDisplayStatus(status);
+  return (
+    <Card className="mb-5 flex-fill" key={booking.id}>
+      <div className="embed-responsive embed-responsive-4by3">
+        <div className="embed-responsive-item">
+          <CardImg className="h-100" top src={booking.listing.listingPicUrl} alt="Listing Cover Photo" />
+        </div>
+      </div>
+      <CardBody>
+        <CardTitle className="h5 font-weight-normal mb-3">{booking.listing.title}</CardTitle>
+        <CardSubtitle className="small mb-3">
+          {addressLine1 && formatAddress(addressLine1, addressLine2, city, state, country, postalCode)}
+          {!addressLine1 && formatGeolocationAddress({ lat, lng, city, country })}
```

Comment:
Small change, but I'd do this as:

    {addressLine1 ? formatAddress(...) : formatGeolocationAddress(...)}

Rationale is maintainability; the logic for the two checks is necessarily complementary (we want exactly one of these formatted outputs to be displayed) but separating them allows them to diverge (e.g if we change `addressLine1` to `(addressLine1 || addressLine2)` on one line but not the next)

Also, I may be misremembering, but I think we added a `fullAddress` field to listings on the backend that handles this formatting ahead of time, so you might just be able to say `booking.listing.fullAddress` here.

Reply:
Yep, backend defines `fullAddress` as:

      fullAddress: {
        type: DataTypes.VIRTUAL,
        get: function () {
          const { addressLine1, addressLine2, city, state, country, postalCode, lat, lng } = this;
          return addressLine1 ?
            formatAddress(addressLine1, addressLine2, city, state, country, postalCode) :
            formatGeolocationAddress({ lat, lng, city, country });
        }
      }

May need to update GraphQL queries to expose it tho; in the GraphQL `booking` end point there is a `Listing` type defined which does not include this.

Reply:
interesting. definitely on board with the maintainability factor (leaves smaller margin of error as stated)

As for the `fullAddress`, I think some cards should only show city, state, country, etc. (omitting street address since it shouldn't be shown to unapproved bookings). Would it be better practice to handle that in the backend?

Reply:
actually lemme smoke test this

Reply:
> As for the `fullAddress`, I think some cards should only show city, state, country, etc. (omitting street address since it shouldn't be shown to unapproved bookings). Would it be better practice to handle that in the backend?

Ah, you're right; `fullAddress` (as currently defined) should only be exposed for approved bookings. Sounds like we want something like a `displayAddress` that takes into account the privileges of the user doing the viewing? In any case, that's getting out of scope for this issue.

Reply:
ahh makes sense. Will put in the fix for a ternary at the very least